src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs

```diff
@@ -815,87 +761,79 @@ private static unsafe string JoinCore<T>(char* separator, int separatorLength, I
             }
         }
 
-        private static unsafe string JoinCore(char* separator, int separatorLength, ReadOnlySpan<string?> value)
+        private static string JoinCore(ReadOnlySpan<char> separator, ReadOnlySpan<string?> values)
         {
-            // If the separator is null, it is converted to an empty string before entering this function.
-            // Even for empty strings, fixed should never return null (it should return a pointer to a null char).
-            Debug.Assert(separator != null);
-            Debug.Assert(separatorLength >= 0);
-
-            int count = value.Length;
-            if (count <= 1)
+            if (values.Length <= 1)
             {
-                return count == 0 ?
-                    string.Empty :
-                    value[0] ?? string.Empty;
+                return values.Length == 0 ?
```

Comment:
Does JIT optimize this? I think we do some trickery in `Span<T>.IsEmpty` in order to allow the following deref into index 0 to have its bounds check elided.

Reply:
You're right.  Thanks.  Looks like that's tracked by https://github.com/dotnet/runtime/issues/10950.